### PR TITLE
Change layer color + export doodles with their colors

### PR DESCRIPTION
**Pull request by theowiik**

### CODE REVIEW

File: src/main/java/chalmers/pimp/model/canvas/LayerUpdateEvent.java [hunk 1]
```diff
@@ -118,6 +120,30 @@ public int getAmountOfLayers() {
...
+  /**
+   * Sets the the selected layer by its index. Does nothing if the index is outside the range: [0,
+   * amount of layers).
+   *
+   * @param index the index of the selected layer.
+   */
+  void setSelectedLayer(int index) {
+    int i = 0;
+    for (IReadOnlyLayer layer : layers) {
+      if (i++ == index) {
```
Comment (oscaralmstrom) [inline src/main/java/chalmers/pimp/model/canvas/LayerUpdateEvent.java]: Might make this more readable :neckbeard:

Comment (albin-johansson) [reply src/main/java/chalmers/pimp/model/canvas/LayerUpdateEvent.java]: This.

File: src/main/java/chalmers/pimp/model/canvas/LayerManager.java [hunk 2]
```diff
@@ -365,6 +369,22 @@ boolean isLayerVisible(int layerIndex) { // TODO test
...
+   * Sets the color of the active layer if the active layer is colorable.
+   *
+   * @param color the color to be set. Does nothing if the color is {@code null}.
+   */
+  void setActiveLayerColor(IColor color) {
+    if (color == null) {
+      return;
+    }
+
+    var layer = getActiveLayer();
```
Comment (Minus98) [inline src/main/java/chalmers/pimp/model/canvas/LayerManager.java]: make sure that the type is explicitly stated when using "var"

File: src/main/java/chalmers/pimp/model/canvas/LayerUpdateEvent.java [hunk 3]
```diff
@@ -138,9 +138,10 @@ public IReadOnlyLayer getSelectedLayer() {
   void setSelectedLayer(int index) {
     int i = 0;
     for (IReadOnlyLayer layer : layers) {
-      if (i++ == index) {
+      if (index == i) {
         selectedLayer = layer;
       }
+      i++;
```
Comment (oscaralmstrom) [inline src/main/java/chalmers/pimp/model/canvas/LayerUpdateEvent.java]: Would love some doc right here. Might be hard for someone less intelligent to understand this :neckbeard:

Comment (albin-johansson) [reply src/main/java/chalmers/pimp/model/canvas/LayerUpdateEvent.java]: Speak for yourself.

Comment (oscaralmstrom) [reply src/main/java/chalmers/pimp/model/canvas/LayerUpdateEvent.java]: Oof

**Pull request merged**